### ISSUE TITLE
WCAG SC 2.5.3 failures not identified by axe DevTools

### ISSUE THREAD

**Issue by theerebuss** [dequelabs/axe-core#4381]
### Product

axe Extension

### Product Version

Extension: v4.77.1 axe-core: v4.8.4

### Latest Version

- [x] I have tested the issue with the latest version of the product

### Issue Description

I was testing some content for WCAG 2.1 SC 2.5.3 and noticed that my example wasn't being flagged by axe DevTools. To confirm the issue, I created [a page to test this in a CodeSandbox](https://codesandbox.io/p/sandbox/wcag-2-5-3-axe-devtools-test-cc6ccq?file=%2Findex.html) using [all the Fail examples](https://dequeuniversity.com/rules/axe/4.8/label-content-name-mismatch?application=RuleDescription) for the rule. I ran the extension on the [example page](https://cc6ccq.csb.app/) and got 0 issues, except for 2 issues caused by the CodeSandbox injected button.
To make sure the button wasn't the cause, I created a page locally to test it and got the same result, no issues with the page.

#### Expectation
label-content-name-mismatch rule fails on 3 separate elements

#### Actual
0 issues found

#### How to Reproduce

1. Open [CodeSandbox example page](https://cc6ccq.csb.app/) or use the code shared below to test locally
2. Run axe DevTools full page scan
3. Observe 0 issues found

For future context, in case the CodeSandbox gets deleted, here is the used code:

<details>

<summary>Sample page that should produce WCAG 2.5.3 failures</summary>

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <title>WCAG SC 2.5.3 test</title>
  </head>
  <body>
    <main>
      <h1>Examples of failures for WCAG SC 2.5.3 Label in Name</h1>

      <div role="link" aria-label="OK">Next</div>

      <button name="link" aria-label="the full">The full label</button>

      <button type="button" aria-label="definitely not WCAG 2.5.3 compliant">
        mysterious button
      </button>
    </main>
  </body>
</html>
```

</details>

#### Additional context

Below are 2 screenshots, with alt text, of the 0 issues in axe DevTools and a successful failure identification by a different accessibility checker:

![A page with the sample code provided is rendered and on the side, the axe DevTools extension is shown, with 0 total errors and the message "You have (0) automatic issues, nice!".](https://github.com/dequelabs/axe-core/assets/39736248/eff3e22b-1969-441e-800d-dc14919d7bee)

![A page with the sample code provided is rendered and on the side, the IBM Equal Access Accessibility Checker is shown, with 3 errors with the message "Accessible name does not match or contain the visible label text". The first one is highlighted and there is a focus indicator on the page on the first rendered button.](https://github.com/dequelabs/axe-core/assets/39736248/49cbdef0-5422-4992-9389-1ef84c5de406)

**Comment by straker** [comment]
Thanks for the issue. So the `label-content-name-mismatch` rule is an [experimental rule](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#experimental-rules), which the extension does not run by default. If you're a Pro user you can enable the experimental rules in the settings and then you should be able to see the results back. Running the example with experimental rules turned on, I see 3 violations returned.

**Comment by straker** [comment]
I can see from the docs that we claim it's enabled in the browser extension, but this is both correct and incorrect. It _was_ enabled in the axe coconut extension (when that was still a thing), but it is not enabled in the devtools extension. We'll update the docs to make that more clear.

**Comment by andrefcdias** [comment]
Thank you @straker! That makes much more sense now.
